### PR TITLE
ref(discover) Remove DiscoverSavedQuery store

### PR DESCRIPTION
**Pull request by markstory**
We no longer have the need to link a menu item and page together with reflux. Additionally we're planning on adding pagination to saved queries which further complicates the reflux store.

### CODE REVIEW

**Review by dashed — APPROVED** [review]
Looks good to me 👍

File: src/sentry/static/sentry/app/components/asyncComponent.tsx [hunk 1]
```diff
@@ -224,7 +224,7 @@ export default class AsyncComponent<
 
   remountComponent = () => {
     if (this.shouldReload) {
-      this.setState({reloading: true}, this.fetchData);
+      this.reloadData();
```
Comment (leedongwei) [inline src/sentry/static/sentry/app/components/asyncComponent.tsx]: `this.reloadData` sets the state but does not call `this.fetchData`. Is removing the callback intended?

Comment (leedongwei) [inline src/sentry/static/sentry/app/components/asyncComponent.tsx]: this.reloadData sets the state but does not call this.fetchData. Is removing the callback intended?

Comment (markstory) [reply src/sentry/static/sentry/app/components/asyncComponent.tsx]: The `reloadData()` method calls `this.fetchData({reloading: true});` which should set state and then fetch.

Comment (leedongwei) [reply src/sentry/static/sentry/app/components/asyncComponent.tsx]: Oh yea, my bad 😬

**Review by leedongwei — APPROVED** [review]

**Pull request merged**